### DISCUSSION TITLE
x-on shorthand does not work with custom prefix

### DISCUSSION THREAD

**Discussion by doompadee** [alpinejs/alpine#2520] [7. Bugs (archived)]
### Discussed in https://github.com/alpinejs/alpine/discussions/2186

<div type='discussions-op-text'>

<sup>Originally posted by **doompadee** October  6, 2021</sup>
The shorthand notation for the `x-on` directive does not work when a custom attribute prefix is configured.

With

```javascript
Alpine.prefix("data-x-");
```
event handlers fail:

```html
<button
   data-x-cloak
   data-x-show="nextBtn"
   @click="revealNext()">Next</button>
```

This is a pity as the extended syntax is not HTML5 compliant (see https://html.spec.whatwg.org/#embedding-custom-non-visible-data-with-the-data-*-attributes).</div>

**Comment by ekwoka** [comment]
This seems intentional and appropriate to me.

The point of using the `data` prefix is mostly because just `x-directive` is not html compliant.

Thus `@` and `:` shorthand are non-compliant as well.

The real question would be what is the point of using a different prefix and also using the shorthands?

**Comment by doompadee** [comment]
@ekwoka As I said: compliance with the HTML spec. Browsers do the right thing, but the grammar checker still complains during building.

As a suitable workaround has been presented in the discussion, this issue might be closed if desired.
